src/process.rs

```diff
@@ -0,0 +1,340 @@
+use std::io;
+use std::path::Path;
+use std::process::{Child, Command, Stdio};
+use sysinfo::{Pid, ProcessesToUpdate, System};
+
+/// Represents a Syncthing process that can be managed by the application.
+///
+/// This struct handles both processes started by the app and external processes.
+/// Uses sysinfo for cross-platform process management.
+/// Tracks all Syncthing processes (parent and children) for complete shutdown.
+pub struct SyncthingProcess {
+    child: Option<Child>,
+    pub started_by_app: bool,
+    pub syncthing_path: String,
+    pub pid: Option<u32>,   // Main process ID
+    tracked_pids: Vec<u32>, // All Syncthing process IDs (parent + children)
+    system: System,         // sysinfo System instance for process monitoring
+}
+
+// Mark SyncthingProcess as safe to send and share between threads
+// This is safe because we've carefully handled Windows handles using usize
+unsafe impl Send for SyncthingProcess {}
+unsafe impl Sync for SyncthingProcess {}
+
+impl SyncthingProcess {
+    /// Creates a new SyncthingProcess instance.
+    pub fn new(path: &str) -> Self {
+        Self {
+            syncthing_path: path.to_string(),
+            child: None,
+            started_by_app: false,
+            pid: None,
+            tracked_pids: Vec::new(),
+            system: System::new(),
+        }
+    }
+    /// Detects if a Syncthing process is currently running and creates a SyncthingProcess instance.
+    pub fn detect_process(syncthing_path: &str, external_only: bool) -> io::Result<Option<Self>> {
+        use sysinfo::{ProcessesToUpdate, System};
+        let mut system = System::new();
+        // Only refresh what we need - just process names and PIDs
+        system.refresh_processes(ProcessesToUpdate::All, false);
+
+        // Get the executable name to search for
+        let exe_name = Path::new(syncthing_path)
+            .file_name()
+            .and_then(|name| name.to_str())
+            .unwrap_or("syncthing");
+
+        // Remove .exe extension if present for cross-platform compatibility
+        let exe_name = exe_name.strip_suffix(".exe").unwrap_or(exe_name);
```

Comment:
Why this is needed? Processes could be just identified by the  `syncthing` part? Also, the function below ( `find_all_syncthing_processes`) is almost a duplicate of this one. Maybe these could be combined?